cl/recap/mergers.py

```diff
@@ -1535,9 +1537,27 @@ async def merge_attachment_page_data(
     if pacer_case_id:
         params["docket_entry__docket__pacer_case_id"] = pacer_case_id
     try:
-        main_rd = await RECAPDocument.objects.select_related(
-            "docket_entry", "docket_entry__docket"
-        ).aget(**params)
+        if is_acms_attachment:
+            # Recap documents on ACMS attachment pages share the same pacer_case_id
+            # which causes an issue when using the aget method. Since the aget
+            # method expects a unique identifier to retrieve a specific document,
+            # utilizing it in this scenario would inevitably result in the
+            # MultipleObjectsReturned exception.
+            #
+            # An alternative approach is to employ the filter method in conjunction
+            # with the afirst method. This combination allows for efficient retrieval
+            # of the main RD (record) of a docket entry.
+            main_rd = (
+                await RECAPDocument.objects.select_related(
+                    "docket_entry", "docket_entry__docket"
+                )
+                .filter(**params)
+                .afirst()
```

Comment:
Are you sure the first will be the desired RD? Note that the `RecapDocument` model has this Meta attribute `ordering = ("document_type", "document_number", "attachment_number")`

Reply:
> Are you sure the first will be the desired RD?

Yes, we use attachment #1 as the `main_rd` for attachment pages. Unlike district court pages, ACMS pages only have attachments. Here's a comparison: 

| District court | ACMS |
| ------------- | ------------- |
| ![image](https://github.com/freelawproject/courtlistener/assets/55959657/3a287116-caba-4d30-a949-bff1bbb472f9)  | ![image](https://github.com/freelawproject/courtlistener/assets/55959657/a05b4f3a-2e84-4ec3-b3cb-6427720889af)  |

>  Note that the RecapDocument model has this Meta attribute ordering = ("document_type", "document_number", "attachment_number")

This won't be an issue because ACMS documents within an attachment page share identical `pacer_doc_id`, `document_type` (ATTACHMENT), `document_number` (the entry number), this ensures the queryset is ordered with attachment #1 appearing first.

| pacer_doc_id | document_type | document_number | attachment_number
| ------------- | ------------- |------------- | ------------- |
| 583c6255-3fd2-ed11-b596-001dd80701be | 2 | 29 | 1 |
| 583c6255-3fd2-ed11-b596-001dd80701be | 2 | 29 | 2 |